### ISSUE TITLE
Taking over change history

### ISSUE THREAD

**Issue by alerque** [liuchengxu/vim-clap#661]
- OS: Arch Linux
 - (Neo)Vim version: 0.4.0
 - vim-clap version: current HEAD

Sometime since I last updated vim-plugins (up to a month perhaps?), the behavior of vim-clap has regressed.

A common workflow for me is to open a file, search for some next, make a change, then use vim-clap with a grep or files provider to load up another file with related content. My expectation is that I should be able to use `n` without further ado to find the next instance from my previous search, then `.` to repeat the fix.

At some point `vim-clap` has started registering things typed into the search (for example using the files provider) as an action in VIM's command history. Now when I get a new buffer open `n` works to find the next instance of my last search, but when I try to repeat the last action using `.` I get whatever fuzzy string I typed into the files provider inserted in my file.

**Comment by liuchengxu** [comment]
It can be helpful if you bisect the commit that introduces this change.

**Comment by alerque** [comment]
Yes, and if I did that I could probably fix it myself too. Trouble is I don't have the bandwidth today or in the next couple days. I may get to it eventually if nobody else does first.
